src/services/milestones/milestones.hooks.js

```diff
@@ -61,6 +62,107 @@ const restrict = () => (context) => {
 };
 
+
+/**
+ *
+ * Conditionally sends a notification after patch or create
+ *
+ **/
+const sendNotification = () => (context) => {
+  const { id, data, app, result } = context;
+
+  /**
+   * Notifications when a milestone get created
+   **/
+  if(context.method === 'create') {
+    if(result.status === 'proposed') {
+      // find the campaign admin and send a notification that milestone is proposed
+      app.service('users').find({query: { address: data.campaignOwnerAddress }})
+        .then((users) => {
+          console.log('campaignOwner', users)
+
+          Notifications.milestoneProposed(app, {
+            recipient: users.data[0].email,
+            user: users.data[0].name,
+            milestoneTitle: data.title,
+            campaignTitle: result.campaign.title,
+            amount: data.maxAmount
+          });  
+        })
+        .catch((e) => console.error('error sending proposed milestone notification', e));
+    };  
+  }
+
+  /**
+   * Notifications when a milestone get patches
+   **/
+  if(context.method === 'patch') {
+    if(result.prevStatus === 'proposed' && result.status === 'InProgress' && result.mined) {
+      // find the milestone owner and send a notification that his/her proposed milestone is approved
+      Notifications.proposedMilestoneAccepted(app, {
+        recipient: result.owner.email,
+        user: result.owner.name,
+        milestoneTitle: result.title,
+        amount: result.maxAmount,
+        txHash: result.txHash
+      });    
+    }
+
+    if(result.prevStatus === 'proposed' && result.status === 'rejected') {
+      // find the milestone owner and send a notification that his/her proposed milestone is rejected
+      Notifications.proposedMilestoneRejected(app, {
+        recipient: result.owner.email,
+        user: result.owner.name,
+        milestoneTitle: result.title,
+        campaignTitle: result.campaign.title        
+      });    
+    } 
+
+    if(result.status === 'NeedsReview') {
+      console.log('result>>>', result);
+      // find the milestone reviewer owner and send a notification that this milestone is been marked as complete and needs review
+      Notifications.milestoneRequestReview(app, {
+        recipient: result.reviewer.email,
+        user: result.reviewer.name,
+        milestoneTitle: result.title,
+        campaignTitle: result.campaign.title
+      });    
+    }  
+
+    // TO DO: Not sure about this one
+    if(result.status === 'Completed') {
+      // find the milestone owner and send a notification that his/her milestone is marked complete
+      Notifications.milestoneMarkedCompleted(app, {
+        recipient: result.owner.email,
+        user: result.owner.name,
+        milestoneTitle: result.title,
+        campaignTitle: result.campaign.title        
+      });    
+    } 
+
+    if(result.prevStatus === 'NeedsReview' && result.status === 'inProgress') {
```

Comment:
I thin this needs to be `result.status === 'InProgress'`

Reply:
No, the status is InProgress as soon as the milestone is created. But we specifically check here for a milestone that moved from needsReview back to inProgress, which is a review rejected milestone.

Reply:
the difference is the capital 'I'. There is not an 'inProgress' status. It is 'InProgress'